tools/hugepages-machineconfig-generator/hugepages-machineconfig-generator.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+
+	"github.com/ghodss/yaml"
+
+	performancev2 "github.com/openshift-kni/performance-addon-operators/api/v2"
+	"github.com/openshift-kni/performance-addon-operators/pkg/utils/hugepages"
+)
+
+const defaultNodeRole = "worker"
+
+var (
+	nodeRole   = flag.String("n", defaultNodeRole, "node role of the machine config object")
+	inputFile  = flag.String("i", "", "performance profile yaml file")
+	outputFile = flag.String("o", "", "performance profile yaml file")
+)
+
+func main() {
+	//parse teh command line flags
+	flag.Parse()
+
+	var err error
+	var bytes []byte
+	if inputFile == nil || *inputFile == "" {
+		//TODO get stdin content as a file
+		bytes, err = ioutil.ReadAll(os.Stdin)
+		if err != nil {
+			log.Fatalf("failed read the input content: %v", err)
+		}
+	} else {
+		//parse the provided file content
+		bytes, err = ioutil.ReadFile(*inputFile)
+		if err != nil {
+			log.Fatalf("failed read the input file: %v", err)
+		}
+	}
+
+	profile := &performancev2.PerformanceProfile{}
+	err = yaml.Unmarshal(bytes, profile)
+	if err != nil {
+		log.Fatalf("failed to unmarshal the input into performance profile: %v", err)
+	}
+
+	mc, err := hugepages.New(profile.Spec.HugePages, *nodeRole)
+	if err != nil {
+		log.Fatalf("failed to generate a machine config with hugepages settings: %v", err)
+	}
+
+	y, err := yaml.Marshal(mc)
+	if err != nil {
+		log.Fatalf("failed to get the machine config as yaml file: %v", err)
+	}
+
+	manifest := string(y)
+
+	if outputFile == nil || *outputFile == "" {
+		//print to stdout console - println also prints to the standart output
+		fmt.Println(manifest)
+	} else {
+		//save mc in the provided yaml file
+		var f *os.File
+		f, err := os.OpenFile(*outputFile, os.O_WRONLY, os.ModeAppend)
+		if err != nil {
+			log.Printf("error opening %s: %s\ncreate the file to save the output\n", *outputFile, err)
+			f, err = os.Create(*outputFile)
```

Comment:
if `OpenFile` succeeds (which you don't need to use btw, you can just use `os.Create` once) you don't need to run `os.Create` again
```
} else {
    f, err := os.Create()...
    if err != nil {
         log.Fatalf(...)
   }
   defer f.Close() // but strictly speaking this can fail as well, but it's so rare we don't bother
   _, err := f.WriteString()
    if err != nil {
         log.Fatalf(...)
   }
```

Reply:
alternatively (untested)
```
var sink io.Writer = os.Stdout
if outputFile != nil && *outputFile != "" {
	f, err := os.Create(*outputFile)
	if err != nil {
		log.Printf("error opening %s: %v\n", *outputFile, err)
	}
	defer f.Close()
	sink = f
}
_, err := io.WriteString(sink, manifest)
if err != nil {
	log.Fatalf("unable to write the output: %v", err)
}
```

Reply:
looks much better, I'll adjust/test that, thanks!